crates/infra/cli/src/commands/setup/mod.rs

```diff
@@ -37,9 +43,32 @@ impl OrderedCommand for SetupCommand {
         Terminal::step(format!("setup {name}", name = self.clap_name()));
 
         match self {
+            SetupCommand::ShellCompletions => generate_shell_completions(),
             SetupCommand::Cargo => setup_cargo(),
             SetupCommand::Npm => setup_npm(),
             SetupCommand::Pipenv => setup_pipenv(),
         }
     }
 }
+
+fn generate_shell_completions() -> Result<()> {
+    println!("Generating shell completions...\n");
+
+    let target_dir = std::env::var("CARGO_TARGET_DIR").unwrap_or_else(|_| "target".to_string());
+    let shell_completions_dir = PathBuf::from(target_dir).join("shell-completions");
+
+    std::fs::create_dir_all(&shell_completions_dir)?;
+
+    let mut cmd = crate::commands::Cli::command();
+    let bin_name = cmd.get_bin_name().expect("Set by #[command]").to_owned();
+
+    for shell in [Shell::Bash, Shell::Zsh] {
+        let file = clap_complete::generate_to(shell, &mut cmd, &bin_name, &shell_completions_dir)?;
+
+        let canonicalized = file.canonicalize()?;
+        let stripped = canonicalized.strip_repo_root()?;
+        println!("Generated file at {}", stripped.display());
```

Comment:
nit:

```suggestion
        println!("Generated {shell} completions file at {}", stripped.display());
```

Reply:
Done in https://github.com/NomicFoundation/slang/pull/739/commits/a189f3900d53326ff3e27bbdb149210287063747